src/mqtt-client/client.js

```diff
@@ -28,66 +28,66 @@ const client = mqtt.connect({
 let blockPublishing = false,
     publishingQueue = new Set();
 
-client.on('connect', onConnect);
+client.on('connect', onMqttConnect);
+client.on('message', onMqttMessage);
 
-function onConnect() {
-    client.subscribe(`${MQTT_INPUT_TOPIC_PREFIX}#`, onSubscribed);
-
-    output.stream.subscribe(event => {
-        let message, topic = MQTT_OUTPUT_TOPIC_PREFIX;
+function onMqttConnect() {
+    client.subscribe(`${MQTT_INPUT_TOPIC_PREFIX}#`);
+    output.stream.subscribe(onInputEvent);
+}
 
-        if ( event.event === DEVICE_STATUS_EVENT) {
-            message = event.value;
-            topic += event.device;
-        } else  if (event.event === DEVICE_INFO_EVENT) {
+function onMqttMessage(topic, rawMessage) {
+    debug(`MQTT >>. Got message: topic '${topic}', message: '${rawMessage.toString()}'`);
+    let message = '';
+    try {
+        message = JSON.parse(rawMessage);
+    } catch (e) {
+        message = rawMessage.toString();
+    }
+    let isDeviceInfo = typeof message === 'object';
+    let event,
+        deviceName = topic.split('/').pop();
 
-            publishDeviceInfoEvent(event.device);
-            return;
+    if (isDeviceInfo) {
 
-        } else {
-            debug(`Unknown output stream event received: '${JSON.stringify(event)}'`);
-            return;
-        }
+        publishingQueue.delete(deviceName);
+        blockPublishing = false;
+        publishDeviceInfoEvent();
 
-        client.publish(topic, message);
-    });
+        event  = {
+            event: 'device-info',
+            device: deviceName,
+            value: message
+        };
+    }
+    else {
+        event = {
+            event: 'status',
+            device: deviceName,
+            value: rawMessage.toString()
+        };
+    }
+    input.write(event);
+    debug(`Added to input stream event: '${JSON.stringify(event)}'`);
 }
 
-function onSubscribed() {
-    client.on('message', function (topic, rawMessage) {
-        debug(`MQTT >>. Got message: topic '${topic}', message: '${rawMessage.toString()}'`);
-        let message = '';
-        try {
-            message = JSON.parse(rawMessage);
-        } catch (e) {
-            message = rawMessage.toString();
-        }
-        let isDeviceInfo = typeof message === 'object';
-        let event,
-            deviceName = topic.split('/').pop();
+function onInputEvent(event) {
+    let message, topic = MQTT_OUTPUT_TOPIC_PREFIX;
 
-        if (isDeviceInfo) {
+    if ( event.event === DEVICE_STATUS_EVENT) {
```

Comment:
I just extracted this code into function, I was not intend to refactor all this stuff around